src/main/java/com/es/phoneshop/web/ProductListAddCartItemServlet.java

```diff
@@ -0,0 +1,57 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.exception.OutOfStockException;
+import com.es.phoneshop.service.CartService;
+import com.es.phoneshop.service.impl.CartServiceImplement;
+import jakarta.servlet.ServletConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.text.NumberFormat;
+import java.text.ParseException;
+
+public class ProductListAddCartItemServlet extends HttpServlet {
+
+    protected CartService cartService;
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        this.cartService = CartServiceImplement.getInstance();
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String quantityStr = request.getParameter("quantity").trim();
+        Long productId = Long.parseLong(request.getParameter("productId").trim());
+
+        if (!quantityStr.matches("[\\d.,]+") || quantityStr.startsWith("0")) {
```

Comment:
Please move quantity validation to separate method and reuse it to avoid duplication + same for parsing quantity